Source/Renderer/ComputeEngine.js

```diff
@@ -0,0 +1,210 @@
+/*global define*/
+define([
+        '../Core/BoundingRectangle',
+        '../Core/Color',
+        '../Core/ComponentDatatype',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/destroyObject',
+        '../Core/DeveloperError',
+        '../Core/Geometry',
+        '../Core/GeometryAttribute',
+        '../Core/PrimitiveType',
+        '../Shaders/ViewportQuadVS',
+        './BufferUsage',
+        './ClearCommand',
+        './DrawCommand',
+        './Framebuffer',
+        './RenderState',
+        './ShaderProgram',
+        './VertexArray'
+    ], function(
+        BoundingRectangle,
+        Color,
+        ComponentDatatype,
+        defaultValue,
+        defined,
+        defineProperties,
+        destroyObject,
+        DeveloperError,
+        Geometry,
+        GeometryAttribute,
+        PrimitiveType,
+        ViewportQuadVS,
+        BufferUsage,
+        ClearCommand,
+        DrawCommand,
+        Framebuffer,
+        RenderState,
+        ShaderProgram,
+        VertexArray) {
+    "use strict";
+
+    /**
+     * @private
+     */
+    var ComputeEngine = function(context) {
+        this._context = context;
+        this._viewportQuadVertexArray = undefined;
+    };
+
+    var viewportQuadAttributeLocations = {
+        position : 0,
+        textureCoordinates : 1
+    };
+    var renderStateScratch;
+    var drawCommandScratch = new DrawCommand({
+        primitiveType : PrimitiveType.TRIANGLES
+    });
+    var clearCommandScratch = new ClearCommand({
+        color : new Color(0.0, 0.0, 0.0, 0.0)
+    });
+
+    function getViewportQuadVertexArray(computeEngine) {
+        var vertexArray = computeEngine._viewportQuadVertexArray;
+
+        if (!defined(vertexArray)) {
+            var geometry = new Geometry({
```

Comment:
We have similar code [here](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Renderer/Context.js#L921).  Perhaps we should make a `@private` `ViewportQuadGeometry` in `Core` and use it in both places?  For a simple-ish example geometry, see [BoxGeometry](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/BoxGeometry.js).